RecoParticleFlow/PFProducer/src/PFMuonAlgo.cc

```diff
@@ -14,138 +14,49 @@ using namespace std;
-  else
-    ptFactorHighEta_ = 2.;
-
-  if(iConfig.exists("metFactorForFakes"))
-    metFactorFake_ = iConfig.getParameter<double>("metFactorForFakes");
-  else
-    metFactorFake_ = 4.;
-
-  if(iConfig.exists("minMomentumForPunchThrough"))
-    minPunchThroughMomentum_ = iConfig.getParameter<double>("minMomentumForPunchThrough");
-  else
-    minPunchThroughMomentum_=100.;
-
-  if(iConfig.exists("minEnergyForPunchThrough"))
-    minPunchThroughEnergy_ = iConfig.getParameter<double>("minEnergyForPunchThrough");
-  else
-    minPunchThroughEnergy_ = 100.;
-
-  if(iConfig.exists("punchThroughFactor"))
-    punchThroughFactor_ = iConfig.getParameter<double>("punchThroughFactor");
-  else
-    punchThroughFactor_ = 3.;
-
-  if(iConfig.exists("punchThroughMETFactor"))
-    punchThroughMETFactor_ = iConfig.getParameter<double>("punchThroughMETFactor");
-  else
-    punchThroughMETFactor_ = 4.;
+    template<class T>
+    T getParameter(edm::ParameterSet const& pset, std::string && name, T && defaultValue) {
+        return pset.exists(name) ? pset.getParameter<T>(name) : defaultValue;
+    }
 
-  if(iConfig.exists("cosmicRejectionDistance"))
-    cosmicRejDistance_ = iConfig.getParameter<double>("cosmicRejectionDistance");
-  else
-    cosmicRejDistance_ = 1.0;
 }
 
+PFMuonAlgo::PFMuonAlgo(const edm::ParameterSet& iConfig)
+
+  : pfCosmicsMuonCleanedCandidates_(std::make_unique<reco::PFCandidateCollection>())
+  , pfCleanedTrackerAndGlobalMuonCandidates_(std::make_unique<reco::PFCandidateCollection>())
+  , pfFakeMuonCleanedCandidates_(std::make_unique<reco::PFCandidateCollection>())
+  , pfPunchThroughMuonCleanedCandidates_(std::make_unique<reco::PFCandidateCollection>())
+  , pfPunchThroughHadronCleanedCandidates_(std::make_unique<reco::PFCandidateCollection>())
+  , pfAddedMuonCandidates_(std::make_unique<reco::PFCandidateCollection>())
+
+  , maxDPtOPt_(getParameter<double>(iConfig, "maxDPtOPt", 1.0))
+  , minTrackerHits_(getParameter<int>(iConfig, "minTrackerHits", 8))
+  , minPixelHits_(getParameter<int>(iConfig, "minPixelHits", 1))
+  , trackQuality_(reco::TrackBase::qualityByName(getParameter<std::string>(iConfig, "trackQuality", "highPurity")))
+  , errorCompScale_(getParameter<double>(iConfig, "ptErrorScale", 4.0))
+  , eventFractionCleaning_(getParameter<double>(iConfig, "eventFractionForCleaning", 0.75))
+  , dzPV_(getParameter<double>(iConfig, "dzPV", 0.2))
+  , postCleaning_(getParameter<bool>(iConfig, "postMuonCleaning", false)) // disable by default (for HLT)
+  , minPostCleaningPt_(getParameter<double>(iConfig, "minPtForPostCleaning", 20.))
+  , eventFactorCosmics_(getParameter<double>(iConfig, "eventFactorForCosmics", 10.))
+  , metSigForCleaning_(getParameter<double>(iConfig, "metSignificanceForCleaning", 3.))
+  , metSigForRejection_(getParameter<double>(iConfig, "metSignificanceForRejection", 4.))
+  , metFactorCleaning_(getParameter<double>(iConfig, "metFactorForCleaning", 3.))
```

Comment:
this was default of 4 before.

Reply:
Right! Thanks for the catch. I was double checking it myself, but missed that one..

Reply:
I changed this line in a fixup to the previous commit, in order to not bloat the repo with too many small fixup commits.